### PR TITLE
[ISSUE-474]fix bug in load data with exclude_dictionary and empty dimensions

### PR DESCRIPTION
**Pull request by eason-lyx**
in surrogate generate step,when change null value to null string ,currently using [meta.memberMapping.length] which is dim.length+measure.length to loop the row.
but row size = dim.length+measure.length+nodictionary.length.
currently process logic will result in null point exception when using dictionary_exclude.
change using row size to loop the row.

### CODE REVIEW

**Comment by jackylk** [comment]
@eason-lyx can you add a test case for it using dictionary_exclude

**Comment by eason-lyx** [comment]
retest this please

**Comment by eason-lyx** [comment]
@jackylk testcase is added.
please review and check in.

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestDataWithDicExcludeAndInclude.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+  }
+
+  override def beforeAll {
+    buildTestData
+    buildTable
+  }
+
+  test("test load data with dictionary exclude & include and with empty dimension") {
+    loadTable
```
Comment (jackylk) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestDataWithDicExcludeAndInclude.scala]: please do a query to test the table is correctly loaded and compare the query test against the expected result.

File: processing/src/main/java/org/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java [hunk 2]
```diff
@@ -833,8 +833,6 @@ private String getCarbonLocalBaseStoreLocation() {
         if (obj.equals(valueToCheckAgainst)) {
           rowValue[i] = CarbonCommonConstants.MEMBER_DEFAULT_VAL;
         }
-      } else if (i > meta.memberMapping.length) {
```
Comment (gvramana) [inline processing/src/main/java/org/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java]: 1. valueToCheckAgainst is never initialized.
2. This comparision applies to only dimension fields  i.e.,dictionary+nondictionary+complextypes_members
@kumarvishal09 @nareshpr please comment on this

Comment (eason-lyx) [inline processing/src/main/java/org/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java]: @gvramana @kumarvishal09 @nareshpr 
1.is valueToCheckAgainst alway use null?
2.the comparision i think we applies to all fields.
   1)if we give only part of fileds,we will get null when processing surrogate step.
   2)meausre is null ,we set to "@NU#LL$!" here. and when handle measure column at following process,we will set measure to null if the value of measure is "@NU#LL$!".and according the datatype give the measure 0L or 0.0.

**Comment by eason-lyx** [comment]
retest this please

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestDataWithDicExcludeAndInclude.scala (hunk 1 above)
Comment (eason-lyx) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestDataWithDicExcludeAndInclude.scala]: ok,already added a query.

**Comment by gvramana** [comment]
retest this please

**Pull request merged**